tests/sentry/snuba/test_discover.py

```diff
@@ -1389,7 +1389,7 @@ def test_enviroment_promoted_tag(self):
         result = discover.get_facets("", params)
         keys = {r.key for r in result}
         assert keys == {"environment", "level"}
-        assert {"prod", "staging", None} == {f.value for f in result if f.key == "environment"}
+        assert {None, "prod", "staging"} == {f.value for f in result if f.key == "environment"}
```

Comment:
@markstory what do you want to do when tags are not present ?
Individual queries return an empty string for the rows that do not have the tag. The aggregate query does not. Are the bars in the tag facet supposed to include a section for the records that do not have the tag or is the bar showing the distribution of the events that have the tag ?

Reply:
An empty string should be ok. I can double check but I think the ui will fold '' into the other segment.

Reply:
Sorry, I think my comment may not have been clear. 
The "tag is not present" group in the query answer will not be there at all. It provides you the distribution of the tags when they exist. If you group by on an arrayJoin of tags, there is no row when a tag does not exists.  Which is the same behavior of any query that groups by tags_key/tags_value
What is the expected behavior / does the product have a specific expectation here ?

How do we calculate the size of "others" today? if the total count for a tag is `count() ...group by tags_key` that would not include records that do not have the tag either.

Reply:
> What is the expected behavior / does the product have a specific expectation here ?

The others and n/a buckets are somewhat similar in how they are presented to users. The other value represents all values outside the top 9. For example if a tag has 100 possible values, the other slice represents the values from the bottom 91 values.

The size of others is inferred based on the total number of events that match the current non-aggregate conditions, subtract the sum of the top 9 values.